frame/tx-pause/src/tests.rs

```diff
@@ -0,0 +1,235 @@
+			full_name::<Test>(b"Balances", b"transfer"),
+		));
+		assert_ok!(call_transfer(4, 1).dispatch(RuntimeOrigin::signed(0)));
+	});
+}
+
+#[test]
+fn can_filter_balance_in_batch_when_paused() {
+	new_test_ext().execute_with(|| {
+		let batch_call =
+			RuntimeCall::Utility(pallet_utility::Call::batch { calls: vec![call_transfer(1, 1)] });
+
+		assert_ok!(TxPause::pause(
+			RuntimeOrigin::signed(mock::PauseOrigin::get()),
+			full_name::<Test>(b"Balances", b"transfer"),
+		));
+
+		assert_ok!(batch_call.clone().dispatch(RuntimeOrigin::signed(0)));
+		System::assert_last_event(
+			pallet_utility::Event::BatchInterrupted {
+				index: 0,
+				error: frame_system::Error::<Test>::CallFiltered.into(),
+			}
+			.into(),
+		);
+	});
+}
+
+#[test]
+fn can_filter_balance_in_proxy_when_paused() {
+	new_test_ext().execute_with(|| {
+		assert_ok!(TxPause::pause(
+			RuntimeOrigin::signed(mock::PauseOrigin::get()),
+			full_name::<Test>(b"Balances", b"transfer"),
+		));
+
+		assert_ok!(Proxy::add_proxy(RuntimeOrigin::signed(1), 2, ProxyType::JustTransfer, 0));
+
+		assert_ok!(Proxy::proxy(RuntimeOrigin::signed(2), 1, None, Box::new(call_transfer(1, 1))));
+		System::assert_last_event(
+			pallet_proxy::Event::ProxyExecuted {
+				result: DispatchError::from(frame_system::Error::<Test>::CallFiltered).into(),
+			}
+			.into(),
+		);
+	});
+}
+
+// GENERAL FAIL/NEGATIVE TESTS ---------------------
+
+#[test]
+fn fails_to_pause_self() {
+	new_test_ext().execute_with(|| {
+		assert_noop!(
+			TxPause::pause(
+				RuntimeOrigin::signed(mock::PauseOrigin::get()),
+				full_name::<Test>(b"TxPause", b"pause"),
+			),
+			Error::<Test>::Unpausable
+		);
+	});
+}
+
+#[test]
+fn fails_to_pause_unpausable_pallet() {
+	new_test_ext().execute_with(|| {
+		assert_noop!(
+			TxPause::pause(
+				RuntimeOrigin::signed(mock::PauseOrigin::get()),
+				full_name::<Test>(b"Balances", b"transfer_keep_alive"),
+			),
+			Error::<Test>::Unpausable
+		);
+	});
+}
+
+#[test]
+fn fails_to_pause_unpausable_call_when_other_call_is_paused() {
+	new_test_ext().execute_with(|| {
+		assert_ok!(call_transfer(1, 1).dispatch(RuntimeOrigin::signed(0)));
+
+		let batch_call =
+			RuntimeCall::Utility(pallet_utility::Call::batch { calls: vec![call_transfer(1, 1)] });
+		assert_ok!(batch_call.clone().dispatch(RuntimeOrigin::signed(0)));
+
+		assert_ok!(TxPause::pause(
+			RuntimeOrigin::signed(mock::PauseOrigin::get()),
+			full_name::<Test>(b"Balances", b"transfer"),
+		));
+
+		assert_ok!(call_transfer_keep_alive(3, 1).dispatch(RuntimeOrigin::signed(3)));
+		assert_err!(
+			call_transfer(2, 1).dispatch(RuntimeOrigin::signed(0)),
+			frame_system::Error::<Test>::CallFiltered
+		);
+	});
+}
+
+#[test]
+fn fails_to_pause_unpausable_call() {
+	new_test_ext().execute_with(|| {
+		assert_noop!(
+			TxPause::pause(
+				RuntimeOrigin::signed(mock::PauseOrigin::get()),
+				full_name::<Test>(b"Balances", b"transfer_keep_alive"),
+			),
+			Error::<Test>::Unpausable
+		);
+	});
+}
```

Comment:
I think this is the same as the `fails_to_pause_unpausable_pallet` test. Maybe we want seperate tests for checking unpausable pallets and unpausable calls.

Reply:
Good catch!  
The code had whole pallet pausing logic in there at one point. Removed the duplicate.